### DISCUSSION TITLE
Extracting Collections and Related

### DISCUSSION THREAD

**Discussion by stefanopagliari** [retorquere/zotero-better-bibtex#1845] [Q&A] [answered]
Could you plese help in figuring what script should I add to the postscript to export the folder/collection in which an item is located and the "related" items?

Thank you

**Comment by retorquere** [comment, marked as answer]
The collections are available as `item.collections` which is an array of keys; the details of the collections are available as `Translator.collections`, addressable by key. If you want all collection names in the field `collections` you'd do

```
reference.add({ name: 'collections', value: item.collections.map(key => Translator.collections[key].name), sep: ', ' })
```

but since collections always carry all their collection keys, but only the collections being exported are actually exposed to the translator, you need to check whether they exist, and it'd become

```
reference.add({ name: 'collections', value: item.collections.map(key => Translator.collections[key] && Translator.collections[key].name).filter(name => name), sep: ', ' })
```

related items can be found in `item.relations`, which is an array of URIs which end in the key of the item. Picking the item out from that is currently more roundabout than it should be, if you open an issue I can address that.

**Reply by stefanopagliari** [reply]
Thank you. 
Currently it is not working for me. Here is what I have currently in my postscript:

```
if (Translator.BetterTeX) {
  reference.add({ name: 'keywords', value: item.tags, sep: ',  ', enc: 'tags' });
  reference.add({ name: 'dateadded', value: item.dateAdded });
  reference.add({ name: 'datemodified', value: item.dateModified });
  reference.add({ name: 'collections', value: item.collections.map(key => translator.collections[key] && translator.collections[key].name).filter(name => name) })

}
```

Sorry, I'm not sure how to troubleshoot this

**Reply by retorquere** [reply]
you have `translator` where it should be `Translator`

**Reply by stefanopagliari** [reply]
Thank you. This works now!

**Reply by retorquere** [reply]
You can choose a different separator like `; ` by using `sep: '; '`

**Reply by stefanopagliari** [reply]
I have noticed that the data about the collections appears only when the entire library is exported, but not if I export a collection. This is not a deal-breaker for me, so I don't want to waste your time, but I was wondering if there is any easy workaround

I'm using this line in the postscript
`reference.add({ name: 'collections', value: item.collections.map(key => Translator.collections[key] && Translator.collections[key].name).filter(name => name), sep: ', ' })
`

**Reply by retorquere** [reply]
There is no workaround. The data that is handed to export translators never includes information about the "root level" of the export - if you export the library, exporters get all collections but can't tell the full library is being exported, if you export a collection you get all subcollections but not the collection being exported, and you can't tell inside a export translator which scenario is in play.
